x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -182,11 +211,44 @@ public DownsampleIndexerAction.ShardDownsampleResponse execute() throws IOExcept
             );
         }
 
-        task.setRollupShardIndexerStatus(RollupShardIndexerStatus.COMPLETED);
+        task.updatePersistentTaskState(
+            new RollupShardPersistentTaskState(RollupShardIndexerStatus.COMPLETED, null),
+            ActionListener.wrap(response -> task.setRollupShardIndexerStatus(RollupShardIndexerStatus.COMPLETED), e -> {
+                logger.error("error while updating downsampling persistent task state", e);
+                throw new ElasticsearchException("error while updating downsampling persistent task state", e);
+            })
+        );
 
         return new DownsampleIndexerAction.ShardDownsampleResponse(indexShard.shardId(), task.getNumIndexed());
     }
 
+    private Query createQuery() throws IOException {
+        if (this.state.started() && this.state.tsid() != null) {
+            final SearchExecutionContext searchExecutionContext = indexService.newSearchExecutionContext(
+                indexShard.shardId().id(),
+                0,
+                searcher,
+                System::currentTimeMillis,
+                null,
+                Collections.emptyMap()
+            );
+            return new BoolQueryBuilder().filter(new TermQueryBuilder(TimeSeriesIdFieldMapper.NAME, this.state.tsid()))
```

Comment:
I think we can use Lucene query api instead of Elasticsearch's `QueryBuilder` api. That way there is no need to create a `SearchExecutionContext` here. For example: 

```
return new BooleanQuery.Builder().add(new TermQuery(new Term(TimeSeriesIdFieldMapper.NAME, this.state.tsid())), BooleanClause.Occur.FILTER).build();
```

Reply:
Actually this query will only match with documents that only have the specified tsid. I think we want to match with all documents that have the specified tsid and beyond? So I think we need a range query? If that is the case we need `SortedSetDocValuesField.newSlowRangeQuery(...)` because the _tsid field isn't indexed.

Reply:
Hey, just a heads up, we've got this speedy query built for sorted indices that could totally work here too:
[link](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/lucene/queries/SearchAfterSortedDocQuery.java). 
Adapting it to our needs shouldn't be a big deal. And if we find it handy, we might even think about adding it directly into Lucene, kind of like what's been done with IndexSortSortedSetDocValuesRangeQuery.

Reply:
Thanks for pointing this out Jim. The main focus is to get this PR merged sooner than later. But we will definitely change this in a followup change.